code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -235,4 +235,159 @@ proc/flame_radius(radius = 1, turf/T, burn_intensity = 25, burn_duration = 25, b
 	name = "\improper Type 8 WP grenade"
 	desc = "A deadly gas grenade found within the ranks of the UPP. Designed to spill white phosporus on the target. It explodes 2 seconds after the pin has been pulled."
 	icon_state = "grenade_upp_wp"
-	item_state = "grenade_upp_wp"
\ No newline at end of file
+	item_state = "grenade_upp_wp"
+
+/obj/item/explosive/grenade/impact
+	name = "\improper M40 IMDP grenade"
+	desc = "A high explosive contact detonation munition utilizing the standard DP canister chassis. Has a focused blast specialized for door breaching and combating emplacements and light armoured vehicles. Capable of being loaded in the M92 Launcher, or thrown by hand."
+	icon_state = "grenade_impact"
+	item_state = "grenade_impact"
+	hud_state = "grenade_frag"
+	det_time = 40
+	dangerous = TRUE
+	underslug_launchable = TRUE
+
+/obj/item/explosive/grenade/impact/prime()
+	spawn(0)
```

Comment:
why do you need to do this on the next decisecond

Reply:
That's the convention for grenades; I assume it exists for a good reason.